model/fn-execution/src/main/proto/beam_fn_api.proto

```diff
@@ -872,3 +872,25 @@ service BeamFnExternalWorkerPool {
   // Stop the SDK worker.
   rpc StopWorker (StopWorkerRequest) returns (StopWorkerResponse) {}
 }
+
+// Request from runner to SDK Harness asking for its status.
+message WorkerStatusRequest {
+  // (Required) Unique ID identifying this request.
+  string request_id = 1;
+}
+
+// Response from SDK Harness to runner containing the debug related status info.
+message WorkerStatusResponse {
+  // (Required) Unique ID from the original request.
+  string request_id = 1;
+  // (Optional) Error message if exception encountered generating the status response.
+  string error = 2;
+  // (Optional) Status debugging info reported by SDK harness worker.
```

Comment:
You need to describe the format and what needs to be part of the response otherwise SDK and runner authors can't tell what they are supposed to do with this.